language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -478,7 +478,9 @@ data Decoder
     | DecoderObject [(T.Text, Decoder)]
     | DecoderConstant DecoderConstant
     | DecoderRef TypeRef -- ^ Reference to an object with a .decoder() method
-    | DecoderLazy Decoder
+    | DecoderLazy Decoder -- ^ Lazy decoder, we need this to avoid infinite loops
```

Comment:
Where is the `DecoderLazy` actually constructed in the codegen?

Reply:
For variants, that part is fine. The problem is that variants are not the only way to write recursive (terminating) types `Optional`, `List` and `TextMap` give you the same issues and therefore need to be lazy as well. This just seemed like the most appropriate place to document what needs to be lazy.

Reply:
The other option would be to make all record fields lazy but I’m not quite sure around the performance implications of making things lazy so limiting it to where we need it to break recursion seemed like a better choice.

Reply:
makes sense, thanks!